src/coprocessor/dag/executor/selection.rs

```diff
@@ -84,12 +82,12 @@ impl Executor for SelectionExecutor {
         self.count = 0;
     }
 
-    fn collect_statistics_into(&mut self, statistics: &mut Statistics) {
-        self.src.collect_statistics_into(statistics);
-    }
-
-    fn collect_metrics_into(&mut self, metrics: &mut ScanCounter) {
+    fn collect_metrics_into(&mut self, metrics: &mut ExecutorMetrics) {
         self.src.collect_metrics_into(metrics);
+        if self.first_collect {
+            metrics.executor_count.selection += 1;
```

Comment:
I think it's a wrong place to increase `executor_count` in `collect_metrics_into`, at least it is misleading. There are no relationship in this two operations according to their semantics.